modules/misc/nixgl.nix

```
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+let cfg = config.nixGL;
+in {
+  meta.maintainers = [ lib.maintainers.smona ];
+
+  options.nixGL.prefix = lib.mkOption {
+    type = lib.types.str;
+    default = "";
+    example = lib.literalExpression
+      ''"''${inputs.nixGL.packages.x86_64-linux.nixGLIntel}/bin/nixGLIntel"'';
+    description = ''
+      The nixGL command that `lib.nixGL.wrap` should wrap packages with.
+      This can be used to provide libGL access to applications on non-NixOS systems.
+
+      Wrap individual packages like so: `(config.lib.nixGL.wrap <package>)`. The returned package
+      can be used just like the original one, but will have access to libGL. If this option is empty (the default),
+      then `lib.nixGL.wrap` is a no-op. This is useful on NixOS, where the wrappers are unnecessary.
+    '';
+  };
+
+  config = {
+    lib.nixGL.wrap = # Wrap a single package with the configured nixGL wrapper
+      pkg:
+
+      if cfg.prefix == "" then
+        pkg
+      else
+      # Wrap the package's binaries with nixGL, while preserving the rest of
+      # the outputs and derivation attributes.
+        (pkg.overrideAttrs (old: {
+          name = "nixGL-${pkg.name}";
+
+          # Make sure this is false for the wrapper derivation, so nix doesn't expect
+          # a new debug output to be produced. We won't be producing any debug info
+          # for the original package.
+          separateDebugInfo = false;
+
```

```suggestion
          nativeBuildInputs = old.nativeBuildInputs or [ ] ++ [ pkgs.makeWrapper ];
```
Fails with `makeWrapper: command not found` otherwise

---

Interesting, it didn't fail for me during testing, but happy to add this as a failsafe.